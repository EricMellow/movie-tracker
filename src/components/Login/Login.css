* {
  background-color: black;
}

.signUpForms {
  display: flex;
  flex-wrap: wrap;
}
.leftOrTop {
  flex: auto;
}

.rightOrBottom {
  flex: auto;
}

h2 {
  color: gray;
  font-size: 25px;
      margin-left: 30px;
      height: 40px;
      width: 100%;
      text-align: left;

}

h3 {
  text-align: left;
  margin-left: 30px;
  color: grey;

}

.signUpForm {
  display: block;
  float: left;
}

.loginForm {
  display: block;
  float: left;
}