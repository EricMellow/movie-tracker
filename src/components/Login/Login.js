import React, { Component } from 'react';
import { connect } from 'react-redux';

export class Login extends Component {
  constructor(props) {
    super(props);

    this.state = {
      name: '',
      signUpEmail: '',
      signUpPassword: '',
      loginEmail: '',
      loginPassword: ''
    };

  }

  onChangeHandler = (event) => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  }

  signUpSubmitHandler = async (event) => {
    event.preventDefault();
    const url = 'http://localhost:3000/api/users/new';

    const retrievedUserInfo = await this.getUsers();

    if (!retrievedUserInfo.emailMatch) {
      await fetch(url, {
        method: 'POST',
        body: JSON.stringify({
          name: this.state.name,
          email: this.state.signUpEmail,
          password: this.state.signUpPassword
        }),
        headers: {
          'Content-Type': 'application/json'
        }
      }
      );
    } else {
      console.log('Password is in use');
    }
  }

  loginSubmitHandler = async (event) => {
    event.preventDefault();

    const retrievedUserInfo = await this.getUsers();

    if (retrievedUserInfo.passwordEmailMatch) {
      const url = 'http://localhost:3000/api/users/';
      try {
        const response = await fetch(url, {
          method: 'POST',
          body: JSON.stringify({
            email: this.state.loginEmail,
            password: this.state.loginPassword
          }),
          headers: {
            'Content-Type': 'application/json'
          }
        }
        );
        const rawData = await response.json();
        const userData = rawData.data;
      } catch(error) {
        console.log(error);
      }
    } else {
      console.log('Email and Password do not match');
    }
  }

  getUsers = async () => {
    const url = 'http://localhost:3000/api/users/';
    const response = await fetch(url);

    const rawData = await response.json();
    const userData = rawData.data;

    return this.validateUser(userData);
  }

  validateUser = (userData) => {
    let emailMatch;
    let passwordEmailMatch;

    const validUser = userData.find(user => {
      return user.email === (this.state.signUpEmail || this.state.loginEmail);
    });
    
    if (validUser) {
      emailMatch = validUser.email ? true : false;
      passwordEmailMatch = (validUser.password === (this.state.signUpPassword || this.state.loginPassword)); 
    }

    return {passwordEmailMatch, emailMatch};
  }
  

  render() {
    return (

      <section className='signUpForms'>

        <article>
          <h2>Sign Up</h2>
          <form
            className='signUpForm'
            onSubmit={this.signUpSubmitHandler}
          >
          <h3>Full Name</h3>
            <input
              name='name'
              value={this.state.name}
              onChange={this.onChangeHandler}
            />
            <h3>Email</h3>
            <input
              name='signUpEmail'
              value={this.state.signUpEmail}
              onChange={this.onChangeHandler}
            />
            <h3>Password</h3>
            <input
              name='signUpPassword'
              value={this.state.signUpPassword}
              onChange={this.onChangeHandler}
            />
            <button>Sign Up</button>
          </form>
        </article>

        <article>
          <h2>Login</h2>
          <form
            className='loginForm'
            onSubmit={this.loginSubmitHandler}
          >
            <h3>Email</h3>
            <input
              name='loginEmail'
              value={this.state.loginEmail}
              onChange={this.onChangeHandler}
            />
            <h3>Password</h3>
            <input
              name='loginPassword'
              value={this.state.loginPassword}
              onChange={this.onChangeHandler}
            />
            <button>Login</button>
          </form>
        </article>

      </section>
    );
  }
}

