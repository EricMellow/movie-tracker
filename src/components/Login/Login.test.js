import React from 'react';
import { Login, mapDispatchToProps } from './Login';
import { shallow } from 'enzyme';
import { createMemoryHistory } from "history";

describe('Login', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Login />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the correct default state', () => {
    const expected = {
      name: '',
      signUpEmail: '',
      signUpPassword: '',
      loginEmail: '',
      loginPassword: '',
      emailPasswordMatch: true,
      emailMatch: true
    };

    expect(wrapper.state()).toEqual(expected);
  });

  describe('onChangeHandler', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<Login />);
    });

    it('should set the correct property in state to the value the user inputs', () => {
      const expected = 'test@email.com';
      const mockEvent = {
        target: {
          name: 'signUpEmail',
          value: 'test@email.com'
        }
      };

      wrapper.find('input[name="signUpEmail"]').simulate('change', mockEvent);

      expect(wrapper.state('signUpEmail')).toEqual(expected);
    });
  });

  describe('signUpSubmitHandler', () => {
    let wrapper;
    let mockEvent;

    beforeEach(() => {
      
      const mockStoreUserId = jest.fn();
      const history = createMemoryHistory('/');
      wrapper = shallow(<Login
        storeUserId={mockStoreUserId}
        history={history}
      />);

      mockEvent = {
        preventDefault: jest.fn()
      };

      window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
        json: () => Promise.resolve({
          data:
            { id: 5 }
        })
      }));

      wrapper.instance().getUsers = jest.fn(()=>([{}]));
      wrapper.instance().validateEmail = jest.fn();
    });

    it('should call fetch with correct arguments', async () => {
      wrapper.instance().getUsers = jest.fn().mockImplementation(() => ({
        emailMatch: false
      }));
      const mockState = {
        name: 'Namebo',
        signUpEmail: 'test@email.com',
        signUpPassword: 'YTho1!'
      };
      
      const expectedUrl = 'http://localhost:3000/api/users/new';
      const expectedOptionsObject = {
        method: 'POST',
        body: JSON.stringify({
          name: 'Namebo',
          email: 'test@email.com',
          password: 'YTho1!'
        }),
        headers: {
          'Content-Type': 'application/json'
        }
      };

      wrapper.setState(mockState);
      await wrapper.instance().signUpSubmitHandler(mockEvent);

      expect(window.fetch).toHaveBeenCalledWith(expectedUrl, expectedOptionsObject);
    });

    it('should call storeNewUser', async ()=>{
      wrapper.instance().storeNewUser = jest.fn();
      const result = wrapper.instance().storeNewUser;
      await wrapper.instance().signUpSubmitHandler(mockEvent);
      
      expect(result).toHaveBeenCalled();
    });

    it('sets emailMatch to false in state if no match is found', async ()=>{
      wrapper.instance().validateEmail = jest.fn().mockImplementation(()=>true);
      await wrapper.instance().signUpSubmitHandler(mockEvent);

      expect(wrapper.state('emailMatch')).toEqual(false);
    });

  });

  describe('loginSubmitHandler', () => {
    let wrapper;
    let mockEvent;

    beforeEach(() => {
      wrapper = shallow(<Login />);
      wrapper.instance().getUsers = jest.fn().mockImplementation(() => ({
        passwordEmailMatch: true
      }));
      wrapper.instance().validateLogin = jest.fn().mockImplementation(() => ({})); 
      mockEvent = {
        preventDefault: jest.fn()
      };
      wrapper.instance().loadExistingUser = jest.fn();
      window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
        status: 200,
        json: () => Promise.resolve({
          id: 1,
          name: "Bread Butterson",
          password: "Toast1!",
          email: "test@gmail.com"
        })
      }));
    });

    it('should call fetch with correct arguments', async () => {
      const mockState = {
        loginEmail: 'test@email.com',
        loginPassword: 'YTho1!'
      };
      const expectedUrl = 'http://localhost:3000/api/users/';
      const expectedOptionsObject = {
        method: 'POST',
        body: JSON.stringify({
          email: 'test@email.com',
          password: 'YTho1!'
        }),
        headers: {
          'Content-Type': 'application/json'
        }
      };

      wrapper.setState(mockState);

      await wrapper.instance().loginSubmitHandler(mockEvent);

      expect(window.fetch).toHaveBeenCalledWith(expectedUrl, expectedOptionsObject);
    });

    it('should call loadExistingUser', async ()=>{
      await wrapper.instance().loginSubmitHandler(mockEvent);

      expect(wrapper.instance().loadExistingUser).toHaveBeenCalled();
    });

    it('should set the state of emailPasswordMatch to false if ther is no match', async ()=>{
      wrapper.instance().validateLogin = jest.fn().mockImplementation(() => (false)); 
      await wrapper.instance().loginSubmitHandler(mockEvent);
      
      expect(wrapper.state('emailPasswordMatch')).toEqual(false);
    });
  });

  describe('getUsers', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<Login />);
    });

    it('should call fetch with the correct argument', async () => {
      wrapper.instance().validateUser = jest.fn();
      const expectedUrl = 'http://localhost:3000/api/users/';


      window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
        json: () => Promise.resolve({})
      }));
      await wrapper.instance().getUsers();

      expect(window.fetch).toHaveBeenCalledWith(expectedUrl);
    });

    it('returns the correct user data', async () => {
      const expected = {
        id: 2,
        name: 'Brett Bretterson',
        password: 'testpass123',
        email: 'test@test.com'
      };

      window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
        json: () => Promise.resolve({ data: expected })
      }));
      const result = await wrapper.instance().getUsers();

      expect(result).toEqual(expected);
    });

  });

  describe('getUserId', () => {
    let wrapper;

    
    beforeEach(()=>{
      wrapper = shallow(<Login />);
      wrapper.setState({
        signUpPassword: 'password',
        signUpEmail: 'test@test.com'
      });
      window.fetch = jest.fn().mockImplementation(()=>Promise.resolve({json: ()=>Promise.resolve({data: {id: 2}})}))
    });

    it.only('should call fetch with the correct arguments', async () => {
      await wrapper.instance().getUserId()
      const expectedOptions = {
        method: 'POST',
        body: JSON.stringify({
          email: 'test@test.com',
          password: 'password'
        }),
        headers: {
          'Content-Type': 'application/json'
        }
      }
      const expectedURL = 'http://localhost:3000/api/users/';

      expect(window.fetch).toHaveBeenCalledWith(expectedURL, expectedOptions)
      
    });

    it('should behave...', () => {
      
    });
  });

  describe('validateUser', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<Login />);
    });

    it('should return an object with a key of emailMatch and value of true if the email is found', () => {
      const mockUserData = [{
        email: 'test@testmail.com',
        password: 'password'
      }];
      wrapper.setState({
        signUpEmail: 'test@testmail.com',
        signUpPassword: '12345'
      });
      const expected = {
        passwordEmailMatch: false,
        emailMatch: true
      };

      const result = wrapper.instance().validateUser(mockUserData);

      expect(result).toEqual(expected);
    });

    it('should return an object with a key of emailMatch and a value of true and a key of passwordEmail match and a value of true if the email and password match', () => {
      const mockUserData = [{
        email: 'test@testmail.com',
        password: 'password'
      }];
      wrapper.setState({
        signUpEmail: 'test@testmail.com',
        signUpPassword: 'password'
      });
      const expected = {
        passwordEmailMatch: true,
        emailMatch: true
      };

      const result = wrapper.instance().validateUser(mockUserData);

      expect(result).toEqual(expected);
    });
  });

  describe('mapDispatchToProps', ()=> {

    it('should call dispatch with the correct params on storeUserId', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockAction = {
        type: 'SET_USER_ID',
        userId: 6
      };
      
      mappedProps.storeUserId(6);
  
      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });
  });

});