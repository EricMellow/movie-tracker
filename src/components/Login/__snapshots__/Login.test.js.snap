// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <article
          className="signUp"
>
          <h2>
                    Sign Up
          </h2>
          <form
                    className="signUpForm"
                    onSubmit={[Function]}
          >
                    <h3>
                              Full Name
                    </h3>
                    <input
                              name="name"
                              onChange={[Function]}
                              placeholder="Name"
                              value=""
                    />
                    <h3>
                              Email
                    </h3>
                    <input
                              name="signUpEmail"
                              onChange={[Function]}
                              placeholder="Ex: testman@aol.com"
                              value=""
                    />
                    
                    <h3>
                              Password
                    </h3>
                    <t
                              buttonClassName=""
                              inputClassName=""
                              name="signUpPassword"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onKeyDown={[Function]}
                              placeholder="Password"
                              useVendorStyles={true}
                              value=""
                    />
                    <button>
                              Sign Up
                    </button>
          </form>
</article>,
        <div
          className="formDivider"
>
          <div />
          <span>
                    OR
          </span>
          <div />
</div>,
        <article
          className="login"
>
          <h2>
                    Login
          </h2>
          <form
                    className="loginForm"
                    onSubmit={[Function]}
          >
                    <h3>
                              Email
                    </h3>
                    <input
                              name="loginEmail"
                              onChange={[Function]}
                              placeholder="Ex: testman@aol.com"
                              value=""
                    />
                    <h3>
                              Password
                    </h3>
                    <t
                              buttonClassName=""
                              inputClassName=""
                              name="loginPassword"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onKeyDown={[Function]}
                              placeholder="Password"
                              useVendorStyles={true}
                              value=""
                    />
                    
                    <button>
                              Login
                    </button>
          </form>
</article>,
      ],
      "className": "signUpForms",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Sign Up
</h2>,
            <form
              className="signUpForm"
              onSubmit={[Function]}
>
              <h3>
                            Full Name
              </h3>
              <input
                            name="name"
                            onChange={[Function]}
                            placeholder="Name"
                            value=""
              />
              <h3>
                            Email
              </h3>
              <input
                            name="signUpEmail"
                            onChange={[Function]}
                            placeholder="Ex: testman@aol.com"
                            value=""
              />
              
              <h3>
                            Password
              </h3>
              <t
                            buttonClassName=""
                            inputClassName=""
                            name="signUpPassword"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onKeyDown={[Function]}
                            placeholder="Password"
                            useVendorStyles={true}
                            value=""
              />
              <button>
                            Sign Up
              </button>
</form>,
          ],
          "className": "signUp",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Full Name
</h3>,
                <input
                  name="name"
                  onChange={[Function]}
                  placeholder="Name"
                  value=""
/>,
                <h3>
                  Email
</h3>,
                <input
                  name="signUpEmail"
                  onChange={[Function]}
                  placeholder="Ex: testman@aol.com"
                  value=""
/>,
                "",
                <h3>
                  Password
</h3>,
                <t
                  buttonClassName=""
                  inputClassName=""
                  name="signUpPassword"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Password"
                  useVendorStyles={true}
                  value=""
/>,
                <button>
                  Sign Up
</button>,
              ],
              "className": "signUpForm",
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Full Name",
                },
                "ref": null,
                "rendered": "Full Name",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "name",
                  "onChange": [Function],
                  "placeholder": "Name",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Email",
                },
                "ref": null,
                "rendered": "Email",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "signUpEmail",
                  "onChange": [Function],
                  "placeholder": "Ex: testman@aol.com",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Password",
                },
                "ref": null,
                "rendered": "Password",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "buttonClassName": "",
                  "inputClassName": "",
                  "name": "signUpPassword",
                  "onBlur": [Function],
                  "onChange": [Function],
                  "onKeyDown": [Function],
                  "placeholder": "Password",
                  "useVendorStyles": true,
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sign Up",
                },
                "ref": null,
                "rendered": "Sign Up",
                "type": "button",
              },
            ],
            "type": "form",
          },
        ],
        "type": "article",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div />,
            <span>
              OR
</span>,
            <div />,
          ],
          "className": "formDivider",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "OR",
            },
            "ref": null,
            "rendered": "OR",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              Login
</h2>,
            <form
              className="loginForm"
              onSubmit={[Function]}
>
              <h3>
                            Email
              </h3>
              <input
                            name="loginEmail"
                            onChange={[Function]}
                            placeholder="Ex: testman@aol.com"
                            value=""
              />
              <h3>
                            Password
              </h3>
              <t
                            buttonClassName=""
                            inputClassName=""
                            name="loginPassword"
                            onBlur={[Function]}
                            onChange={[Function]}
                            onKeyDown={[Function]}
                            placeholder="Password"
                            useVendorStyles={true}
                            value=""
              />
              
              <button>
                            Login
              </button>
</form>,
          ],
          "className": "login",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login",
            },
            "ref": null,
            "rendered": "Login",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Email
</h3>,
                <input
                  name="loginEmail"
                  onChange={[Function]}
                  placeholder="Ex: testman@aol.com"
                  value=""
/>,
                <h3>
                  Password
</h3>,
                <t
                  buttonClassName=""
                  inputClassName=""
                  name="loginPassword"
                  onBlur={[Function]}
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Password"
                  useVendorStyles={true}
                  value=""
/>,
                "",
                <button>
                  Login
</button>,
              ],
              "className": "loginForm",
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Email",
                },
                "ref": null,
                "rendered": "Email",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "name": "loginEmail",
                  "onChange": [Function],
                  "placeholder": "Ex: testman@aol.com",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Password",
                },
                "ref": null,
                "rendered": "Password",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "buttonClassName": "",
                  "inputClassName": "",
                  "name": "loginPassword",
                  "onBlur": [Function],
                  "onChange": [Function],
                  "onKeyDown": [Function],
                  "placeholder": "Password",
                  "useVendorStyles": true,
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Login",
                },
                "ref": null,
                "rendered": "Login",
                "type": "button",
              },
            ],
            "type": "form",
          },
        ],
        "type": "article",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <article
            className="signUp"
>
            <h2>
                        Sign Up
            </h2>
            <form
                        className="signUpForm"
                        onSubmit={[Function]}
            >
                        <h3>
                                    Full Name
                        </h3>
                        <input
                                    name="name"
                                    onChange={[Function]}
                                    placeholder="Name"
                                    value=""
                        />
                        <h3>
                                    Email
                        </h3>
                        <input
                                    name="signUpEmail"
                                    onChange={[Function]}
                                    placeholder="Ex: testman@aol.com"
                                    value=""
                        />
                        
                        <h3>
                                    Password
                        </h3>
                        <t
                                    buttonClassName=""
                                    inputClassName=""
                                    name="signUpPassword"
                                    onBlur={[Function]}
                                    onChange={[Function]}
                                    onKeyDown={[Function]}
                                    placeholder="Password"
                                    useVendorStyles={true}
                                    value=""
                        />
                        <button>
                                    Sign Up
                        </button>
            </form>
</article>,
          <div
            className="formDivider"
>
            <div />
            <span>
                        OR
            </span>
            <div />
</div>,
          <article
            className="login"
>
            <h2>
                        Login
            </h2>
            <form
                        className="loginForm"
                        onSubmit={[Function]}
            >
                        <h3>
                                    Email
                        </h3>
                        <input
                                    name="loginEmail"
                                    onChange={[Function]}
                                    placeholder="Ex: testman@aol.com"
                                    value=""
                        />
                        <h3>
                                    Password
                        </h3>
                        <t
                                    buttonClassName=""
                                    inputClassName=""
                                    name="loginPassword"
                                    onBlur={[Function]}
                                    onChange={[Function]}
                                    onKeyDown={[Function]}
                                    placeholder="Password"
                                    useVendorStyles={true}
                                    value=""
                        />
                        
                        <button>
                                    Login
                        </button>
            </form>
</article>,
        ],
        "className": "signUpForms",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Sign Up
</h2>,
              <form
                className="signUpForm"
                onSubmit={[Function]}
>
                <h3>
                                Full Name
                </h3>
                <input
                                name="name"
                                onChange={[Function]}
                                placeholder="Name"
                                value=""
                />
                <h3>
                                Email
                </h3>
                <input
                                name="signUpEmail"
                                onChange={[Function]}
                                placeholder="Ex: testman@aol.com"
                                value=""
                />
                
                <h3>
                                Password
                </h3>
                <t
                                buttonClassName=""
                                inputClassName=""
                                name="signUpPassword"
                                onBlur={[Function]}
                                onChange={[Function]}
                                onKeyDown={[Function]}
                                placeholder="Password"
                                useVendorStyles={true}
                                value=""
                />
                <button>
                                Sign Up
                </button>
</form>,
            ],
            "className": "signUp",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sign Up",
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3>
                    Full Name
</h3>,
                  <input
                    name="name"
                    onChange={[Function]}
                    placeholder="Name"
                    value=""
/>,
                  <h3>
                    Email
</h3>,
                  <input
                    name="signUpEmail"
                    onChange={[Function]}
                    placeholder="Ex: testman@aol.com"
                    value=""
/>,
                  "",
                  <h3>
                    Password
</h3>,
                  <t
                    buttonClassName=""
                    inputClassName=""
                    name="signUpPassword"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Password"
                    useVendorStyles={true}
                    value=""
/>,
                  <button>
                    Sign Up
</button>,
                ],
                "className": "signUpForm",
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Full Name",
                  },
                  "ref": null,
                  "rendered": "Full Name",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "name",
                    "onChange": [Function],
                    "placeholder": "Name",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Email",
                  },
                  "ref": null,
                  "rendered": "Email",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "signUpEmail",
                    "onChange": [Function],
                    "placeholder": "Ex: testman@aol.com",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                "",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Password",
                  },
                  "ref": null,
                  "rendered": "Password",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "buttonClassName": "",
                    "inputClassName": "",
                    "name": "signUpPassword",
                    "onBlur": [Function],
                    "onChange": [Function],
                    "onKeyDown": [Function],
                    "placeholder": "Password",
                    "useVendorStyles": true,
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Sign Up",
                  },
                  "ref": null,
                  "rendered": "Sign Up",
                  "type": "button",
                },
              ],
              "type": "form",
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div />,
              <span>
                OR
</span>,
              <div />,
            ],
            "className": "formDivider",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "OR",
              },
              "ref": null,
              "rendered": "OR",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                Login
</h2>,
              <form
                className="loginForm"
                onSubmit={[Function]}
>
                <h3>
                                Email
                </h3>
                <input
                                name="loginEmail"
                                onChange={[Function]}
                                placeholder="Ex: testman@aol.com"
                                value=""
                />
                <h3>
                                Password
                </h3>
                <t
                                buttonClassName=""
                                inputClassName=""
                                name="loginPassword"
                                onBlur={[Function]}
                                onChange={[Function]}
                                onKeyDown={[Function]}
                                placeholder="Password"
                                useVendorStyles={true}
                                value=""
                />
                
                <button>
                                Login
                </button>
</form>,
            ],
            "className": "login",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login",
              },
              "ref": null,
              "rendered": "Login",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3>
                    Email
</h3>,
                  <input
                    name="loginEmail"
                    onChange={[Function]}
                    placeholder="Ex: testman@aol.com"
                    value=""
/>,
                  <h3>
                    Password
</h3>,
                  <t
                    buttonClassName=""
                    inputClassName=""
                    name="loginPassword"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Password"
                    useVendorStyles={true}
                    value=""
/>,
                  "",
                  <button>
                    Login
</button>,
                ],
                "className": "loginForm",
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Email",
                  },
                  "ref": null,
                  "rendered": "Email",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "name": "loginEmail",
                    "onChange": [Function],
                    "placeholder": "Ex: testman@aol.com",
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Password",
                  },
                  "ref": null,
                  "rendered": "Password",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "buttonClassName": "",
                    "inputClassName": "",
                    "name": "loginPassword",
                    "onBlur": [Function],
                    "onChange": [Function],
                    "onKeyDown": [Function],
                    "placeholder": "Password",
                    "useVendorStyles": true,
                    "value": "",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Login",
                  },
                  "ref": null,
                  "rendered": "Login",
                  "type": "button",
                },
              ],
              "type": "form",
            },
          ],
          "type": "article",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
